x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/EsQueryExec.java

```diff
@@ -108,23 +111,32 @@ public FieldAttribute field() {
         }
     }
 
-    public EsQueryExec(Source source, EsIndex index, IndexMode indexMode, List<Attribute> attributes, QueryBuilder query) {
-        this(source, index, indexMode, attributes, query, null, null, null);
+    public EsQueryExec(
+        Source source,
+        String indexName,
+        IndexMode indexMode,
+        Map<String, IndexMode> indexNameWithModes,
+        List<Attribute> attributes,
+        QueryBuilder query
+    ) {
+        this(source, indexName, indexMode, indexNameWithModes, attributes, query, null, null, null);
     }
 
     public EsQueryExec(
         Source source,
-        EsIndex index,
+        String indexName,
         IndexMode indexMode,
+        Map<String, IndexMode> indexNameWithModes,
```

Comment:
Why is this map needed?

Reply:
It is used in LocalExecutionPlanner:

https://github.com/elastic/elasticsearch/blob/2ff1d769a61a6c4974b4ab3347ec668d5abf0da4/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java#L564-L571

Reply:
If this is the only thing that requires that: I believe this check could happen much earlier. Potentially already in the analyzer on the coordinator node. We shouldn't need to wait until the local planning to determine that an index isn't a lookup index.

Could you maybe check if it's feasible to move that to an earlier place and, thus, remove the `indexNameWithModes` map altogether from the `EsRelation` and related classes?

Happy to assist with moving that to an earlier stage!

Reply:
There already is an index mode check in the Analyzer: https://github.com/elastic/elasticsearch/blob/4ac8d55ff7c5612413e4a45e14a2b39f3e85f3a9/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java#L239-L255

Reply:
I think I got it. The check is indeed already performed and could be removed - it's just there out of defensiveness as far as I can see.

What cannot be removed is the line
```
 var entry = indicesWithModes.entrySet().iterator().next(); 
```
which, in case of aliases, contains the _actual_ lookup index name in the entry's key.

Looks to me like we should add a comment and leave this as-is. (Well, and maybe remove the defensive double validation, but it also doesn't hurt.)

I'll confirm with @idegtiarenko if this is what we want to do.

Reply:
Correct. IndexName could actually be alias or pattern, while map entries include resolved index names

Reply:
@costin , we decided that's too large in scope for this PR. Ievgen opened a follow-up issue: https://github.com/elastic/elasticsearch/issues/120210

Reply:
The map seems similar to EsIndex - it's detailed information that isn't required.
Nit: doing it in a separate PR will require a transport bump; maybe consider this complete but not merge, do the follow-up PR and then merge them at the same time to have only one increment?

Reply:
I am not sure I would be able to pick up followup immediately.
Also keeping this updated required multiple conflict resolutions. I would really prefer to merge this earlier.